### PR TITLE
single container: add prometheus process

### PR DESCRIPTION
**Pull request by uwedeportivo**
Adds prometheus in the single container deployment scenario. Follow-up PR will add grafana.

We should debate if we want to make this optional or not. I'm leaning towards keeping it in, since a future PR will expose frontend UI that shows grafana dashboards.

I have a demo container image if somebody wants to play with this. I wasn't sure where to push the image. Running it requires an additional volume arg for the prometheus data disk.

### CODE REVIEW

**Comment by slimsag** [comment]
I'll review this once I am back from vacation next week :) in the meantime
feel free to grab temporary review from someone else and merge before then.

**Comment by uwedeportivo** [comment]
You can try it with something like this:

`
docker run --publish 7080:7080 --publish 2633:2633  --publish 9090:9090 --rm --volume ~/.sourcegraph/config:/etc/sourcegraph --volume ~/.sourcegraph/data:/var/opt/sourcegraph --volume ~/sourcegraph-docker/prometheus-disk:/prometheus-data sourcegraph/server:3.6.2-prom
`

File: dev/prometheus/prometheus_local.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  - "searcher.rules"
+  - "service.rules"
+  - "symbols.rules"
+
+# A scrape configuration containing exactly one endpoint to scrape for metrics:
+scrape_configs:
+  # Scrape prometheus itself for metrics.
+  - job_name: "prometheus"
+    static_configs:
+      - targets: ["localhost:9090"]
```
Comment (keegancsmith) [inline dev/prometheus/prometheus_local.yml]: @beyang is it safe for us to use localhost? Or is it better to use 127.0.0.1? I can't remember that context so well.

File: dev/prometheus/prometheus_targets_local.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    # searcher
+    - 127.0.0.1:6069
+    # symbols
+    - 127.0.0.1:6071
+    # repo-updater
+    - 127.0.0.1:6074
+    # query-runner
+    - 127.0.0.1:6067
+    # zoekt-indexserver
+    # - 127.0.0.1:6072
```
Comment (keegancsmith) [inline dev/prometheus/prometheus_targets_local.yml]: yeah we have no prom metrics in zoekt (yet).

File: dev/prometheus/prometheus_local.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      # For Linux, use this line:
+      #- targets: ["localhost:9323"]
+      #
+      # For Mac, use this line:
+      #- targets: ["docker.for.mac.host.internal:9323"]
+      #
+      # For Windows, use this line:
+      #- targets: ["docker.for.win.localhost:9090"]
+
+  - job_name: "node"
```
Comment (tsenart) [inline dev/prometheus/prometheus_local.yml]: This name is easy to be confused with the Prometheus [node_exporter](https://github.com/prometheus/node_exporter) (should we also be running this?). Can we name this something else, like, `sourcegraph`?

**Comment by uwedeportivo** [comment]
Thanks for the review guys. In this PR I only wanted to add prometheus to the single container context. I simply took over the configs and prometheus version that we use in the other contexts. Didn't do any cleanup (other than appeasing the linters). There's two (or more) followups I need to address: upgrade our prometheus, de-duping the configs, clean up the configs. There's too much repetition across the deploy matrix.

File: dev/prometheus/prometheus_local.yml (hunk 3 above)
Comment (uwedeportivo) [reply dev/prometheus/prometheus_local.yml]: Yeah, I'll put it on the list of cleanups when I address the config rewrites. Right now this is inherited and like this in the other deploy contexts.

File: cmd/server/Dockerfile [hunk 4]
```diff
@@ -1,3 +1,4 @@
+FROM sourcegraph/prometheus:v1.4.1@sha256:1a02878ac9a0f532a68673bda93d406b21786706446c42a40cb3ed3e867278c0 AS prom
```
[the remark this answers is not in the document]
Comment (uwedeportivo) [reply cmd/server/Dockerfile]: Yep, definitely. Coming up in the config cleanup. Right now this is the version from the other deploy contexts. Will upgrade all of them.

File: dev/prometheus/prometheus_local.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  evaluation_interval: 30s # Evaluate rules every 30s. Default is 1m.
+  # scrape_timeout is set to the global default (10s).
+
+# Alertmanager configuration
+alerting:
+  alertmanagers:
+    - static_configs:
+        - targets:
+          # deploy-sourcegraph-docker does not yet use prometheus alerts
+          # - alertmanager:9093
```
[the remark this answers is not in the document]
Comment (uwedeportivo) [reply dev/prometheus/prometheus_local.yml]: It's a good question, will add to the list of things I need to address.

**Pull request merged**